src/common/ModelMetadata.ts

```diff
@@ -0,0 +1,92 @@
+export class ModelMetadata {
+    constructor(public tableName: string, public attributes: any){}
+
+    public async getAttributesForScan(args: any) : Promise<any>{
+
+        let ExclusiveStartKey, FilterExpression, Limit, ExpressionAttributeValues, ExpressionAttributeNames;
+        
+        if (args["id"]){
+            ExpressionAttributeNames = {};
+            ExpressionAttributeValues = {};
+
+            FilterExpression = `#id = :hashKey`;
+            ExpressionAttributeValues[':hashKey'] = args["id"];
+            ExpressionAttributeNames[`#id`] = 'id';
+        }
+        else{
+            Limit = args["limit"]? (args["limit"] > 0)? args["limit"] : 10 : undefined;
+
+            if (args["lastEvaluatedKey"]){
+                ExclusiveStartKey = { "id": args["lastEvaluatedKey"] };
+            }
+
+            delete args['limit'];
+            delete args['lastEvaluatedKey'];
+
+            if ( Object.keys(args).length > 0){
+                let filterExp = [];
+                ExpressionAttributeValues = {};
+                ExpressionAttributeNames = {};
+
+                Object.keys(args).forEach(key => {
+                    filterExp.push(`contains(#${key}, :${key})`);
+                    ExpressionAttributeValues[`:${key}`] = args[key];
+                    ExpressionAttributeNames[`#${key}`] = key;
+                });
+                FilterExpression = filterExp.join(', ');
+            }
+        }
+        
+        
+        const expression = {
+            FilterExpression,
+            ExpressionAttributeValues,
+            ExpressionAttributeNames,
+            Limit,
+            ExclusiveStartKey
+        };
+
+
+        // console.log(expression);
```

Comment:
How would you handle logging in a production app?

Reply:
Well, I ussually encapsulate the logging decision into a king of "Logging" library/components. Inside it I put the logic to know when, where and how much to log based on the environment, the configure logging sources, sampling mechanisms, etc).

The consumer codes (lambdas or this 'ModelMetadata.ts' file for instance) don't care. They just use the .log, .warn. .error function of it and that's it.